grype/version/semantic_constraint.go

```diff
@@ -50,14 +59,11 @@ func (c semanticConstraint) Satisfied(version *Version) (bool, error) {
 		return true, nil
 	}
 
-	if !c.supported(version.Format) {
-		return false, fmt.Errorf("(semantic) unsupported format: %s", version.Format)
+	if c.checker == nil {
+		return false, fmt.Errorf("check function is undefined")
```

Comment:
is an undefined check function an error? also this error isn't very specific about what is wrong (from a user perspective).

Reply:
In that case it would, because the zero value for a function var is nil, it would panic. I moved away from that with struct embedding and method overriding.

Reply:
> In that case it would, because the zero value for a function var is nil, it would panic.

correct, and though I'm not endorsing this path at all, I was asking is returning an error the right thing to do or should we patch a more sane default when this is found:
```golang
if c.checker == nil {
		c.checker = func(...){}
}
```
preventing the panic and not stopping the constraint checks (unless we really do need this check).